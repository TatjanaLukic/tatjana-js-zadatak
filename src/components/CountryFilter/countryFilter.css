.country-filter {
  display: flex;
  box-sizing: content-box;
}
@media screen and (min-width: 992px) {
  .country-filter {
    justify-content: center;
    align-items: center;
  }
}

/*# sourceMappingURL=countryFilter.css.map */
