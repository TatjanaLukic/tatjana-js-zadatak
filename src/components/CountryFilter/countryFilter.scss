@import "../../assets/variables";

.country-filter {
  display: flex;
  box-sizing: content-box;
  margin: 0.5rem 0.7rem;

  @media screen and (min-width: $largeSize) {
    justify-content: center;
    align-items: center;
  }
}

.country-filter-item {
  background-color: white;
  border: none;
  color: $cf-main-color;
  text-align: center;
  padding: 0.3rem;

  @media screen and (min-width: $smallSize) {
    margin: 0.5rem;
    font-weight: bold;
    width: 30px;
    height: 30px;
  }
}

.country-filter-item--checked {
  color: $cf-active-color;
  background-color: $cf-second-color;
}
