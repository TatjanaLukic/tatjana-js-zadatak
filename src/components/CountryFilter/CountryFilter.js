import React, { useContext, useState, useEffect } from "react";
import { useLocation } from "react-router-dom";
import CountryContext from "../../context/countryContext";
import PropTypes from "prop-types";
import CountryFilterItem from "./CountryFilterItem/CountryFilterItem";

import "./countryFilter.css";

const CountryFilter = () => {
  const country = useContext(CountryContext);
  const [disabled, setDisabled] = useState(false);

  const regex = RegExp("/NewsPage*");
  const location = useLocation();
  useEffect(() => {
    if (regex.test(location.pathname)) {
      setDisabled(true);
    } else {
      setDisabled(false);
    }
  }, [location]);

  const selectCF = (cf) => {
    country.selectCountry(cf);
  };

  return (
    <div className="country-filter">
      <CountryFilterItem
        id="us"
        value="us"
        onChange={() => selectCF("us")}
        checked={country.country === "us"}
        labelName="US"
        className="country-filter-item"
        disabled={disabled}
      />
      <CountryFilterItem
        id="gb"
        value="gb"
        onChange={() => selectCF("gb")}
        checked={country.country === "gb"}
        labelName="GB"
        className="country-filter-item"
        disabled={disabled}
      />
    </div>
  );
};

CountryFilter.propTypes = {
  country: PropTypes.shape({
    country: PropTypes.string.isRequired,
    selectCountry: PropTypes.func.isRequired,
  }),
};

export default CountryFilter;
