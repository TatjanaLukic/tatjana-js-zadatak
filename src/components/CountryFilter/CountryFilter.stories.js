import React from "react";
import CountryFilter from "./CountryFilter";

export default {
  title: "App/CountryFilter",
  component: CountryFilter,
};

const Template = () => <CountryFilter />;

export const Default = Template.bind({});
