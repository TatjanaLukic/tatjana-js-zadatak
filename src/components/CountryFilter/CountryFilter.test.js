import React from "react";
import { render, screen } from "@testing-library/react";
import routeData from "react-router";
import CountryContext from "../../context/countryContext";
import CountryFilter from "./CountryFilter";

describe("<CountryFilter />", () => {
  let countryFilter;

  afterEach(() => {
    jest.resetAllMocks();
  });

  it("should render 2 CountryFilterItem, us checked by default", async () => {
    const mockLocation = {
      pathname: "/TopNewsPage",
      hash: "",
      search: "",
      state: {},
    };
    jest.spyOn(routeData, "useLocation").mockReturnValue(mockLocation);
    render(
      <CountryContext.Provider
        value={{ country: "us", selectCountry: () => {} }}
      >
        <CountryFilter />
      </CountryContext.Provider>
    );
    const countryFilterItems = await screen.findAllByTestId("cf-item");
    expect(countryFilterItems).toHaveLength(2);
    expect(countryFilterItems[0].querySelector("input").value).toBe("us");
    expect(countryFilterItems[0].querySelector("input")).toHaveAttribute(
      "checked"
    );
    expect(countryFilterItems[0].querySelector("input")).not.toHaveAttribute(
      "disabled"
    );
    expect(countryFilterItems[1].querySelector("input").value).toBe("gb");
    expect(countryFilterItems[1].querySelector("input")).not.toHaveAttribute(
      "checked"
    );
    expect(countryFilterItems[1].querySelector("input")).not.toHaveAttribute(
      "disabled"
    );
  });

  it("disabaled for any route containing /NewsPage  ", async () => {
    const mockLocation1 = {
      pathname: "/NewsPage",
      hash: "",
      search: "",
      state: {},
    };
    jest.spyOn(routeData, "useLocation").mockReturnValue(mockLocation1);
    render(
      <CountryContext.Provider
        value={{ country: "us", selectCountry: () => {} }}
      >
        <CountryFilter />
      </CountryContext.Provider>
    );
    const countryFilterItems = await screen.findAllByTestId("cf-item");
    expect(countryFilterItems).toHaveLength(2);

    expect(countryFilterItems[0].querySelector("input")).toHaveAttribute(
      "disabled"
    );
    expect(countryFilterItems[1].querySelector("input")).toHaveAttribute(
      "disabled"
    );
  });

  it("matches snapshoot for active ", async () => {
    const mockLocation = {
      pathname: "/TopNewsPage",
      hash: "",
      search: "",
      state: {},
    };
    jest.spyOn(routeData, "useLocation").mockReturnValue(mockLocation);
    const { container } = render(
      <CountryContext.Provider
        value={{ country: "us", selectCountry: () => {} }}
      >
        <CountryFilter />
      </CountryContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
