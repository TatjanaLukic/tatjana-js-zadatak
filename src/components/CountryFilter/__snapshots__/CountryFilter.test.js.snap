// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryFilter /> matches snapshoot for active  1`] = `
<div>
  <div
    class="country-filter"
  >
    <div
      data-testid="cf-item"
    >
      <input
        checked=""
        hidden=""
        id="us"
        type="radio"
        value="us"
      />
      <label
        class="country-filter-item country-filter-item--checked"
        for="us"
      >
        US
      </label>
    </div>
    <div
      data-testid="cf-item"
    >
      <input
        hidden=""
        id="gb"
        type="radio"
        value="gb"
      />
      <label
        class="country-filter-item"
        for="gb"
      >
        GB
      </label>
    </div>
  </div>
</div>
`;
